library/ssl_cache.c

```diff
@@ -138,24 +138,18 @@ int mbedtls_ssl_cache_get( void *data,
     return( ret );
 }
 
-int mbedtls_ssl_cache_set( void *data,
-                           unsigned char const *session_id,
-                           size_t session_id_len,
-                           const mbedtls_ssl_session *session )
+static int ssl_cache_find_fresh_entry( mbedtls_ssl_cache_context *cache,
```

Comment:
Fixed the name.

Reply:
@mpg Thank you for spotting the logic flaw, I don't know how I could miss that - esp. because I remember thinking about something along those lines. Anyway, I will create an issue for this pre-existing defect.

Reply:
https://github.com/ARMmbed/mbedtls/issues/4509

Reply:
I reconsidered and decided it's probably easier to fix this along the way. I've made a quick attempt in the latest commit, could you take a look?

Reply:
If you disagree that this should be done here, I can revert and move to another PR. Let me know.